### ISSUE TITLE
Do you have any particular reason to require at least 2 input features?

### ISSUE THREAD

**Issue by maxmarketit** [automl/Auto-PyTorch#56]
I found that the mininumu number of features for input is 2. Is there any reason?

  Because I made custom loss function utilizing two input variables asymmetrically,

  and it does not seem to work. So I am wondering if there is any particular reason 

  that might lead to poor training... I might add my particular loss function and the result later

**Comment by LMZimmer** [comment]
Could you specify what you are trying to do? Which data are you working with, with search space etc.?

**Comment by maxmarketit** [comment]
It seems my loss function works as it is expected, but the result is not up to my expectation so you might take a look at it and see if there is anything wrong with the optimization.

I will introduce my objective briefly. I am doing quantile regression with q-x as input and y(q) as output. But since loss is something like max(q(y-t), (q-1)(y-t)), we need to know q in calculating loss, I would let q have direct path to loss when designing the architecture. But I dont know that option when using automl, I put q in target and made modification to the loss, which is `QuantileLoss2` in my code.

Anyhow, my loss works fine as you can see `Quantile-QinputLoss-Copy1.html`.

As I am working this example as a toy example, checking if auto-pytorch suits my problem,
the result of `AutoPyTorch-Quantile-qinput.html` turns out a bit of disappointment,
considering all the time it spent searching ~ 10 hrs I guess.

So the designer and developer of auto-pytorch might see the possible progess somewhere somehow I guess.

Anyway I always thank you for your dedication and commitment![](url)

**Comment by maxmarketit** [comment]
I think the problem is related to [this issue](https://github.com/automl/Auto-PyTorch/issues/55)

  Thanks to your suggestions in [this issue](https://github.com/automl/Auto-PyTorch/issues/33), I succeeded to change the loss but I do not know how to change the scoring metric or the loss for BOHB. Could you explain how to change the scoring metrics and the loss for BOHB? Thank you in advance!

**Comment by ravinkohli** [comment]
Hey, thanks for this question. Actually, we have been working on a new version of autopytorch that is available on the branch [`development](https://github.com/automl/Auto-PyTorch/tree/development), for docs, see [this](https://automl.github.io/Auto-PyTorch/development/).  Now we feel it's ready for people to test their workflows with this version. 

So I suggest you to try out your data with the development version. For an example on how to use the API, find it [here](https://automl.github.io/Auto-PyTorch/development/examples/index.html)

You can specify the metric to be optimized using the `optimize_metric` argument to the `search` function of the task. 
Feel free to ask any questions about the implementation here. And if you feel the issue is resolved, kindly close it.
